app/code/Magento/Catalog/Controller/Adminhtml/Product/Gallery/Upload.php

```diff
@@ -98,6 +98,9 @@ public function execute()
                 $mediaDirectory->getAbsolutePath($this->productMediaConfig->getBaseTmpMediaPath())
             );
 
+            if ($result === false) {
```

Comment:
It is not allowed to throw and catch exception in the same method - https://devdocs.magento.com/guides/v2.4/coding-standards/technical-guidelines.html#5-exceptions
Add` 'error'` message to `$result` instead

Reply:
hi @xmav, replaced as you suggested